src/resources/js/cart.js

```diff
@@ -0,0 +1,95 @@
+class Cart {
+    constructor(productObj) {
+        this.productObj = productObj;
+    }
+    viewProductsInCart() {
+        var userToken = localStorage.getItem('userToken');
+        $.ajax({
```

Comment:
@AlexeiaW let's pull all graphql calls into a service wrapper, which wraps the jquery ajax method and handles authorisation internally. jQuery ajax methods also can return promises, so you won't need to pass callbacks, you can just return the result of the ajax method, and within the app you could chain promises on to it, and use async / await
```
try {
const data = await $.ajax({ ... })
} catch( err ){
  console.error( err )
}
```

The graphql method could handle most things internally, except that you'd want to pass the actual query to it, and possibly variables to replace in the query. The stringification could happen inside the method, along with the setting of the headers etc. 

When you're building out this method we just need to consider how we can mock it in unit testing.

Reply:
Ok I made a service, please check if I understood correctly and let me know @luke-siedle

Reply:
@AlexeiaW that's exactly it. Looks good.